azure-kusto-data/azure/kusto/data/helpers.py

```diff
@@ -1,14 +1,37 @@
 import sys
-from typing import TYPE_CHECKING, Union
+from typing import TYPE_CHECKING, Union, Callable, Dict, Optional
+from functools import lru_cache
+import copy
 
 if TYPE_CHECKING:
-    import pandas
+    import pandas as pd
     from azure.kusto.data._models import KustoResultTable, KustoStreamingResultTable
 
+# Alias for dataframe_from_result_table converter type
+Converter = Optional[Dict[str, Union[str, Callable[[str, "pd.DataFrame"], "pd.Series"]]]]
+
+
+@lru_cache(maxsize=1, typed=False)
+def default_dict() -> Dict[str, Callable[[str, "pd.DataFrame"], "pd.Series"]]:
```

Comment:
Return `Converter`

Reply:
Even though in this case the return type will never be None? (So its not Optional like Converter)

Reply:
Great point!

I think you should define the Converter type without the Optional, and then annotate Optional[Converter] whenever it is optional (it's also clearer)